### PR TITLE
(FACT-1604) update directory precedence and non-root user tests

### PR DESCRIPTION
**Pull request by ericldelaney**
Add TestRail ID
Add risk tag
Add code style formatting updates
(No code changes where made to the tests (besides updating the code style)

Tested on:
arista4-32a, ciscoxr-64a%7Bhypervisor=vmpooler%7D, cisconx-64a%7Bhypervisor=vmpooler%7D, cumulus25-64a, debian8-64a, fedora25-64a, osx1012-64a, redhat7-64a, sles12-64a, solaris11-64a, ubuntu1604-64a, windows10ent-64a, windows2016-64a

### CODE REVIEW

**Comment by kurtwall** [comment]
Could you correct the typo in the git commit message, please (`s/precdence/precedence/`)?

File: acceptance/tests/external_facts/fact_directory_precedence.rb [hunk 1]
```diff
@@ -1,94 +1,112 @@
-test_name "Fact directory precedence and resolution order for facts"
+# Verify that facter uses the new AIO default path for external facts, /opt/puppetlabs/agent/facts.d
+# also for non-root. non-root case might be its own.. case
```
Comment (kurtwall) [inline acceptance/tests/external_facts/fact_directory_precedence.rb]: Stray punctuation or a shorthand I don't understand?

**Comment by ericldelaney** [comment]
@kurtwall typo fixed

File: acceptance/tests/external_facts/fact_directory_precedence.rb [hunk 2]
```diff
@@ -1,94 +1,112 @@
...
+    etc_puppetlabs_factsd_dir = get_etc_puppetlabs_factsd_dir(agent['platform'])
+    factsd_path = "#{factsd_dir}/test.yaml"
+    etc_factsd_path = "#{etc_factsd_dir}/test.yaml"
+    etc_puppetlabs_factsd_path = "#{etc_puppetlabs_factsd_dir}/test.yaml"
+
+    teardown do
+      on(agent, "rm -rf #{factsd_dir}")
+      on(agent, "rm -rf #{etc_factsd_dir}")
+      on(agent, "rm -rf #{etc_puppetlabs_factsd_dir}")
+    end
```
Comment (kurtwall) [inline acceptance/tests/external_facts/fact_directory_precedence.rb]: This is three SSH sessions. I recommend doing this with puppet code:

```
manifest = 'file { ["/tmp/foo","/tmp/bar","/tmp/baz"]: ensure => absent, force => true }'
apply_manifest_on(agent, manifest)
...
```

Or at least in a single `on(...)` expression to eliminate the extra SSH sessions.

File: acceptance/tests/external_facts/fact_directory_precedence.rb (hunk 1 above)
Comment (ericldelaney) [inline acceptance/tests/external_facts/fact_directory_precedence.rb]: Cut and paste from TestRail, I'll fix both

File: acceptance/tests/external_facts/fact_directory_precedence.rb (hunk 2 above)
Comment (ericldelaney) [inline acceptance/tests/external_facts/fact_directory_precedence.rb]: I was avoiding making code changes on these tests (assuming they were good), but I'll go back, review and update them.

Comment (kurtwall) [inline acceptance/tests/external_facts/fact_directory_precedence.rb]: Fair enough.

Comment (ericldelaney) [inline acceptance/tests/external_facts/fact_directory_precedence.rb]: I'm already working on the changes, I'll update the PR when I've happy and have it tested.

**Comment by ericldelaney** [comment]
@kurtwall Reviewed and updated

**Comment by joshcooper** [comment]
@branan can you also merge this (and the previous fix for FACT-1578) up to master?

**Pull request merged**